lib/hammer_cli_foreman/exception_handler.rb

```diff
@@ -147,6 +154,24 @@ def rake_command
 
     private
 
+    def authenticator_error_message(e)
+      case e.type
+      when :negotiate then negotiation_error_message(e)
+      end
+    end
+
+    def negotiation_error_message(e)
+      case e.code
```

Comment:
This may not be always safe. In this particular case, try `kdestroy; hammer auth login negotiate`

```
[DEBUG 2022-05-03T11:33:40 Exception] Using exception handler HammerCLIForeman::ExceptionHandler#handle_general_exception
[ERROR 2022-05-03T11:33:40 Exception] Error: undefined method `code' for #<ApipieBindings::AuthenticatorError:0x0000559ae28ce178>
Error: undefined method `code' for #<ApipieBindings::AuthenticatorError:0x0000559ae28ce178>
[ERROR 2022-05-03T11:33:40 Exception] 

NoMethodError (undefined method `code' for #<ApipieBindings::AuthenticatorError:0x0000559ae28ce178>):
    /usr/share/gems/gems/hammer_cli_foreman-3.3.0.pre.develop/lib/hammer_cli_foreman/exception_handler.rb:165:in `negotiation_error_message'
    /usr/share/gems/gems/hammer_cli_foreman-3.3.0.pre.develop/lib/hammer_cli_foreman/exception_handler.rb:159:in `authenticator_error_message'
    /usr/share/gems/gems/hammer_cli_foreman-3.3.0.pre.develop/lib/hammer_cli_foreman/exception_handler.rb:117:in `handle_authenticator_error'
    /usr/share/gems/gems/hammer_cli-3.3.0.pre.develop/lib/hammer_cli/exception_handler.rb:31:in `handle_exception'
    /usr/share/gems/gems/hammer_cli-3.3.0.pre.develop/lib/hammer_cli/messages.rb:26:in `handle_exception'
    /usr/share/gems/gems/hammer_cli-3.3.0.pre.develop/lib/hammer_cli/abstract.rb:107:in `rescue in run'
    /usr/share/gems/gems/hammer_cli-3.3.0.pre.develop/lib/hammer_cli/abstract.rb:102:in `run'
    /usr/share/gems/gems/clamp-1.1.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /usr/share/gems/gems/clamp-1.1.2/lib/clamp/command.rb:63:in `run'
    /usr/share/gems/gems/hammer_cli-3.3.0.pre.develop/lib/hammer_cli/abstract.rb:103:in `run'
    /usr/share/gems/gems/clamp-1.1.2/lib/clamp/subcommand/execution.rb:11:in `execute'
    /usr/share/gems/gems/clamp-1.1.2/lib/clamp/command.rb:63:in `run'
    /usr/share/gems/gems/hammer_cli-3.3.0.pre.develop/lib/hammer_cli/abstract.rb:103:in `run'
    /usr/share/gems/gems/clamp-1.1.2/lib/clamp/command.rb:132:in `run'
    /usr/share/gems/gems/hammer_cli-3.3.0.pre.develop/bin/hammer:147:in `<top (required)>'
    /usr/bin/hammer:23:in `load'
    /usr/bin/hammer:23:in `<main>'
```

Reply:
Updated both PRs, should :crossed_fingers: be fixed now.